habitat_sim/agent/controls/pyrobot_noisy_controls.py

```diff
@@ -0,0 +1,266 @@
+            ),
+        ),
+        Movebase=ControllerNoiseModel(
+            linear_motion=MotionNoiseModel(
+                _MultivariateGaussian([0.192, 0.117], [0.055, 0.144]),
+                _MultivariateGaussian([0.128], [0.143]),
+            ),
+            rotational_motion=MotionNoiseModel(
+                _MultivariateGaussian([0.002, 0.001], [0.001, 0.001]),
+                _MultivariateGaussian([0.173], [0.025]),
+            ),
+        ),
+    ),
+}
+
+
+@attr.s(auto_attribs=True)
+class PyRobotNoisyActuationSpec(ActuationSpec):
+    r"""Struct to hold parameters for pyrobot noise model
+
+    https://pyrobot.org/
+    https://github.com/facebookresearch/pyrobot
+
+    Please cite PyRobot if you use this noise model
+
+    Args:
+        amount (float): The amount the control moves the scene node by
+        robot (str): Which robot to simulate noise for.  Valid values
+            are LoCoBot and LoCoBot-Lite
+        controller (str): Which controller to simulate noise models for,
+            Valid values are ILQR, Proportional, Movebase
+            ILQR is the default
+        noise_multiplier (float): Multiplier on the noise amount,
+            useful for ablating the effect of noise
+    """
+    robot: str = attr.ib(default="LoCoBot")
+
+    @robot.validator
+    def check(self, attribute, value):
+        assert value in pyrobot_noise_models.keys(), f"{value} not a known robot"
+
+    controller: str = attr.ib(default="ILQR")
+
+    @controller.validator
+    def check(self, attribute, value):
+        assert value in [
+            "ILQR",
+            "Proportional",
+            "Movebase",
+        ], f"{value} not a known controller"
+
+    noise_multiplier: float = 1.0
+
+
+_x_axis = 0
+_y_axis = 1
+_z_axis = 2
+
+
+def _noisy_action_impl(
+    scene_node: hsim.SceneNode,
+    translate_amount: float,
+    rotate_amount: float,
+    multiplier: float,
+    model: MotionNoiseModel,
+):
+    move_ax = (
+        np.array(scene_node.absolute_transformation().rotation_scaling())
+        @ hsim.geo.FRONT
+    )
+    prep_ax = np.cross(move_ax, hsim.geo.UP)
+
+    # + EPS to make sure 0 is positive.  We multiply the mean by the sign of the translation
+    # as otherwise forward would overshoot on average and backward would undershoot, while
+    # both should overshoot
+    translation_noise = multiplier * np.random.multivariate_normal(
+        np.sign(translate_amount + 1e-8) * model.linear.mean, model.linear.cov
+    )
+    scene_node.translate_local(
+        move_ax * (translate_amount + translation_noise[0])
+        + prep_ax * translation_noise[1]
+    )
+
+    # Same deal with rotation about + EPS and why we multiply by the sign
+    rot_noise = multiplier * np.random.multivariate_normal(
+        np.sign(rotate_amount + 1e-8) * model.rotation.mean, model.rotation.cov
+    )
+
+    scene_node.rotate_local(mn.Deg(rotate_amount) + mn.Rad(rot_noise[0]), hsim.geo.UP)
```

Comment:
@mosra -- Strongly typed angles are great!  I can now add degrees and radians without fear :)

Reply:
Hah, didn't even know this was possible (it isn't in C++, there it can't decide on what the result type should be). Nice to see it "just works" in Python.